webapp/src/pages/boardPage.tsx

```diff
@@ -41,25 +39,28 @@ class BoardPage extends React.Component<Props, State> {
     constructor(props: Props) {
         super(props)
 
+        // Backward compatible query param urls to regular urls
         const queryString = new URLSearchParams(window.location.search)
-        let boardId = queryString.get('id') || ''
-        let viewId = queryString.get('v') || ''
+        const queryBoardId = queryString.get('id')
+        const queryViewId = queryString.get('v')
+        if (queryBoardId) {
+            const params = {...props.match.params, boardId: queryBoardId}
+            if (queryViewId) {
+                params.viewId = queryViewId
+            }
+            const newPath = generatePath(props.match.path, params)
+            props.history.push(newPath)
```

Comment:
if we're changing the URL, should we not return at this point?

Reply:
I'm not 100% sure, this is the constructor of the BoardPage Component, that maybe is reused even if the route has changed, so I think the right thing is to entirely initialize the component and let the router decide if something needs to change.

Reply:
oks, happy to refactor this later if it actually becomes a problem